### DISCUSSION TITLE
The macOS codesign fails after PyInstaller migration from 5.1 to 6.2: _internal/Python.framework/Python: bundle format is ambiguous

### DISCUSSION THREAD

**Discussion by avptin** [pyinstaller/pyinstaller#8146] [Help] [answered]
It looks like the issue is related to the symlinks handling by codesign. 
The issue was likely introduced with https://github.com/pyinstaller/pyinstaller/pull/7619

+ pyinstaller --distpath /Users/runner/app --clean --add-data=tmp:tmp ./app.py
INFO: PyInstaller: 6.2.0
INFO: Python: 3.8.13
INFO: Platform: macOS-12.3-x86_64-i386-64bit

+ codesign --force --deep --options=runtime --entitlements /Users/runner/ci/entitlements.plist --verbose --timestamp --sign Developer ID Application: MY ID /Users/runner/app/_internal/Python.framework/Python
/Users/runner/app/_internal/Python.framework/Python: replacing existing signature
/Users/runner/app/_internal/Python.framework/Python: bundle format is ambiguous (could be app or framework)

All suggestions are welcome.

**Comment by rokm** [comment, marked as answer]
This looks like a case of ruined symbolic links.

Try listing contents of `/Users/runner/app/_internal` with `ls -l`. `Python` in there should be a symlink to `Python.framework/Versions/3.X/Python`. 

Then try listing contents of `/Users/runner/app/_internal/Python.framework`. `Python` there should be a symlink to `Versions/Current/Python`.

In your case, I assume they re both hard copies?

`pyinstaller --distpath /Users/runner/app --clean --add-data=tmp:tmp ./app.py` should create `/Users/runner/app/app/{app,_internal}`. But you seem to be signing `/Users/runner/app/_internal` - is there some file movement or copying involved between the two steps that you have shown us? Does that perhaps involve using `cp -r` (which does not preserve symlinks on macOS, in contrast to linux) instead of `cp -R` (which preserves symlinks)?

**Reply by avptin** [reply]
Hi @rokm , thank you so much for the help!
In my case the artifacts where archived with the zip command line tool without symbolic link preservations (no -y parameter used).
Best Regards, Oleksiy
